src/main/java/com/gmail/nossr50/skills/herbalism/HerbalismManager.java

```diff
@@ -757,69 +757,69 @@ private boolean processGreenThumbPlants(BlockState blockState, BlockBreakEvent b
             return false;
         }
 
-        //If the ageable is NOT mature and the player is NOT using a hoe, abort
-
         Player player = getPlayer();
         PlayerInventory playerInventory = player.getInventory();
         Material seed;
 
         switch (blockState.getType().getKey().getKey().toLowerCase(Locale.ROOT)) {
             case "carrots":
-                seed = Material.matchMaterial("CARROT");
+                seed = Material.CARROT;
                 break;
-
             case "wheat":
-                seed = Material.matchMaterial("WHEAT_SEEDS");
+                seed = Material.WHEAT_SEEDS;
                 break;
-
             case "nether_wart":
-                seed = Material.getMaterial("NETHER_WART");
+                seed = Material.NETHER_WART;
                 break;
-
             case "potatoes":
-                seed = Material.matchMaterial("POTATO");
+                seed = Material.POTATO;
                 break;
-
             case "beetroots":
-                seed = Material.matchMaterial("BEETROOT_SEEDS");
+                seed = Material.BEETROOT_SEEDS;
                 break;
-
             case "cocoa":
-                seed = Material.matchMaterial("COCOA_BEANS");
+                seed = Material.COCOA_BEANS;
                 break;
-
             case "torchflower":
                 seed = Material.matchMaterial("TORCHFLOWER_SEEDS");
                 break;
             default:
                 return false;
         }
 
-
         ItemStack seedStack = new ItemStack(seed);
 
         if (ItemUtils.isAxe(blockBreakEvent.getPlayer().getInventory().getItemInMainHand())
-        && blockState.getType() != Material.COCOA) {
+                && blockState.getType() != Material.COCOA) {
             return false;
         }
 
         if (!greenTerra && !ProbabilityUtil.isSkillRNGSuccessful(SubSkillType.HERBALISM_GREEN_THUMB, player)) {
             return false;
         }
 
-        if (!playerInventory.containsAtLeast(seedStack, 1)) {
+        // Check for seeds in both the main hand and off-hand
+        if (!(playerInventory.containsAtLeast(seedStack, 1) || playerInventory.getItemInOffHand().getType() == seed)) {
+            NotificationManager.sendPlayerInformation(player, NotificationType.REQUIREMENTS_NOT_MET, "Skills.NeedMore", StringUtils.getPrettyItemString(seed));
             return false;
         }
 
+        // Remove seed from the appropriate inventory slot (main hand or off-hand)
```

Comment:
This would need to happen after the subSkillEvent to give other plugins a chance to cancel this ability.